Bug when trying to build cnpy

### Steps to reproduce

I run spack install cnpy and it complains that it cannot find a C compiler. If I go to the package.py file, there is no dependency on "c" and the CMakeLists.txt file itself doesn't ask for any C compiler. So not sure why cmake is asking for a C compiler

I am able to fix this issue by putting depends_on("c", type="build")

This is on an m3 MacBook Pro with Sonomo 14.7.6

### Error message

```console
==> No patches needed for cnpy
==> cnpy: Executing phase: 'cmake'
==> [2025-06-06-16:02:02.560034] '/Users/finkeljo/spack/opt/spack/darwin-m1/cmake-3.31.6-722ouavnqffkvlziklwtn7uw4n2fbr6g/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/Users/finkeljo/spack/opt/spack/darwin-m1/cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/Users/finkeljo/spack/opt/spack/darwin-m1/cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/lib;/Users/finkeljo/spack/opt/spack/darwin-m1/cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/lib64' '-DCMAKE_PREFIX_PATH:STRING=/Users/finkeljo/spack/opt/spack/darwin-m1/cmake-3.31.6-722ouavnqffkvlziklwtn7uw4n2fbr6g;/Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy;/Users/finkeljo/spack/opt/spack/darwin-m1/gmake-4.4.1-giwfkogusl3lu3lf2bzq4id52l42lvzk;/Users/finkeljo/spack/opt/spack/darwin-m1/zlib-ng-2.2.4-hroj5litluhy7p2p74v3xdmww3bpizqc' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_FIND_FRAMEWORK:STRING=LAST' '-DCMAKE_FIND_APPBUNDLE:STRING=LAST' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DCMAKE_POLICY_DEFAULT_CMP0042:STRING=NEW' '-DCMAKE_MACOSX_RPATH=ON' '/var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-stage-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-src'
CMake Deprecation Warning at CMakeLists.txt:1 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The C compiler identification is unknown
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy/libexec/spack/case-insensitive/cc
-- Check for working C compiler: /Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy/libexec/spack/case-insensitive/cc - broken
CMake Error at /Users/finkeljo/spack/opt/spack/darwin-m1/cmake-3.31.6-722ouavnqffkvlziklwtn7uw4n2fbr6g/share/cmake-3.31/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy/libexec/spack/case-insensitive/cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/private/var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-stage-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-build-tw5fc6i/CMakeFiles/CMakeScratch/TryCompile-zdQWQD'
    
    Run Build Command(s): /Users/finkeljo/spack/opt/spack/darwin-m1/cmake-3.31.6-722ouavnqffkvlziklwtn7uw4n2fbr6g/bin/cmake -E env VERBOSE=1 /Users/finkeljo/spack/opt/spack/darwin-m1/gmake-4.4.1-giwfkogusl3lu3lf2bzq4id52l42lvzk/bin/gmake -f Makefile cmTC_ac8e0/fast
    /Users/finkeljo/spack/opt/spack/darwin-m1/gmake-4.4.1-giwfkogusl3lu3lf2bzq4id52l42lvzk/bin/gmake  -f CMakeFiles/cmTC_ac8e0.dir/build.make CMakeFiles/cmTC_ac8e0.dir/build
    gmake[1]: Entering directory '/private/var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-stage-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-build-tw5fc6i/CMakeFiles/CMakeScratch/TryCompile-zdQWQD'
    Building C object CMakeFiles/cmTC_ac8e0.dir/testCCompiler.c.o
    /Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy/libexec/spack/case-insensitive/cc   -arch arm64 -o CMakeFiles/cmTC_ac8e0.dir/testCCompiler.c.o -c /private/var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-stage-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-build-tw5fc6i/CMakeFiles/CMakeScratch/TryCompile-zdQWQD/testCCompiler.c
    /Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy/libexec/spack/case-insensitive/cc: line 401: SPACK_CC_LINKER_ARG: ERROR: LINKER ARG WAS NOT SET, MAYBE THE PACKAGE DOES NOT DEPEND ON CC?
    gmake[1]: *** [CMakeFiles/cmTC_ac8e0.dir/build.make:81: CMakeFiles/cmTC_ac8e0.dir/testCCompiler.c.o] Error 1
    gmake[1]: Leaving directory '/private/var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-stage-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-build-tw5fc6i/CMakeFiles/CMakeScratch/TryCompile-zdQWQD'
    gmake: *** [Makefile:134: cmTC_ac8e0/fast] Error 2
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:6 (project)


-- Configuring incomplete, errors occurred!
==> Error: ProcessError: Command exited with status 1:
    '/Users/finkeljo/spack/opt/spack/darwin-m1/cmake-3.31.6-722ouavnqffkvlziklwtn7uw4n2fbr6g/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/Users/finkeljo/spack/opt/spack/darwin-m1/cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/Users/finkeljo/spack/opt/spack/darwin-m1/cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/lib;/Users/finkeljo/spack/opt/spack/darwin-m1/cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/lib64' '-DCMAKE_PREFIX_PATH:STRING=/Users/finkeljo/spack/opt/spack/darwin-m1/cmake-3.31.6-722ouavnqffkvlziklwtn7uw4n2fbr6g;/Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy;/Users/finkeljo/spack/opt/spack/darwin-m1/gmake-4.4.1-giwfkogusl3lu3lf2bzq4id52l42lvzk;/Users/finkeljo/spack/opt/spack/darwin-m1/zlib-ng-2.2.4-hroj5litluhy7p2p74v3xdmww3bpizqc' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_FIND_FRAMEWORK:STRING=LAST' '-DCMAKE_FIND_APPBUNDLE:STRING=LAST' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DCMAKE_POLICY_DEFAULT_CMP0042:STRING=NEW' '-DCMAKE_MACOSX_RPATH=ON' '/var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-stage-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-src'

3 errors found in build log:
     12    -- The C compiler identification is unknown
     13    -- The CXX compiler identification is AppleClang 16.0.0.16000026
     14    -- Detecting C compiler ABI info
     15    -- Detecting C compiler ABI info - failed
     16    -- Check for working C compiler: /Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrd
           ymgmzmy/libexec/spack/case-insensitive/cc
     17    -- Check for working C compiler: /Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrd
           ymgmzmy/libexec/spack/case-insensitive/cc - broken
  >> 18    CMake Error at /Users/finkeljo/spack/opt/spack/darwin-m1/cmake-3.31.6-722ouavnqffkvlziklwtn7uw4n2fbr6g/share/cmake-3.31/M
           odules/CMakeTestCCompiler.cmake:67 (message):
     19      The C compiler
     20    
     21        "/Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy/libexec/spack/case-i
           nsensitive/cc"
     22    
     23      is not able to compile a simple test program.
     24    

     ...

     29        Run Build Command(s): /Users/finkeljo/spack/opt/spack/darwin-m1/cmake-3.31.6-722ouavnqffkvlziklwtn7uw4n2fbr6g/bin/cma
           ke -E env VERBOSE=1 /Users/finkeljo/spack/opt/spack/darwin-m1/gmake-4.4.1-giwfkogusl3lu3lf2bzq4id52l42lvzk/bin/gmake -f M
           akefile cmTC_ac8e0/fast
     30        /Users/finkeljo/spack/opt/spack/darwin-m1/gmake-4.4.1-giwfkogusl3lu3lf2bzq4id52l42lvzk/bin/gmake  -f CMakeFiles/cmTC_
           ac8e0.dir/build.make CMakeFiles/cmTC_ac8e0.dir/build
     31        gmake[1]: Entering directory '/private/var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-sta
           ge-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-build-tw5fc6i/CMakeFiles/CMakeScratch/TryCompile-zdQWQD'
     32        Building C object CMakeFiles/cmTC_ac8e0.dir/testCCompiler.c.o
     33        /Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy/libexec/spack/case-in
           sensitive/cc   -arch arm64 -o CMakeFiles/cmTC_ac8e0.dir/testCCompiler.c.o -c /private/var/folders/4c/gqmm0tnx02j8lghz_kkg
           5jr00011wr/T/finkeljo/spack-stage/spack-stage-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-build-tw5fc6i/CMakeFiles
           /CMakeScratch/TryCompile-zdQWQD/testCCompiler.c
     34        /Users/finkeljo/spack/opt/spack/darwin-m1/compiler-wrapper-1.0-gecb33fqbhojrvmpfeaiottrdymgmzmy/libexec/spack/case-in
           sensitive/cc: line 401: SPACK_CC_LINKER_ARG: ERROR: LINKER ARG WAS NOT SET, MAYBE THE PACKAGE DOES NOT DEPEND ON CC?
  >> 35    gmake[1]: *** [CMakeFiles/cmTC_ac8e0.dir/build.make:81: CMakeFiles/cmTC_ac8e0.dir/testCCompiler.c.o] Error 1
     36        gmake[1]: Leaving directory '/private/var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-stag
           e-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-build-tw5fc6i/CMakeFiles/CMakeScratch/TryCompile-zdQWQD'
  >> 37        gmake: *** [Makefile:134: cmTC_ac8e0/fast] Error 2
     38    
     39    
     40    
     41    
     42    
     43      CMake will not be able to correctly generate this project.

See build log for details:
  /var/folders/4c/gqmm0tnx02j8lghz_kkg5jr00011wr/T/finkeljo/spack-stage/spack-stage-cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x/spack-build-out.txt

==> Warning: Skipping build of pace-2023.11.25.2-lowgntrzjby275tpycf75tcttauzmh2r since cnpy-master-tw5fc6if3yjsridg45uax5kghjqq577x failed
```

### Information on your system

* **Spack:** 1.0.0.dev0 (8234dc8a9526ad302fef8aff780eb55f9d4add79)
* **Python:** 3.9.6
* **Platform:** darwin-sonoma-m1

<- This is the system python. My spack built python is 3.11.11

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

--- a/repos/spack_repo/builtin/packages/cnpy/package.py
+++ b/repos/spack_repo/builtin/packages/cnpy/package.py
@@ -19,7 +19,8 @@ class Cnpy(CMakePackage):
 
     version("master", branch="master")
 
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
 
     depends_on("zlib-api", type="link")
 

